sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -592,6 +612,45 @@ def set_read_only(
         except binascii.Error as exc:
             raise binascii.Error("Connection string secret has incorrect padding") from exc
 
+    @distributed_trace
+    def list_labels(
+        self,
+        *,
+        name: Optional[str] = None,
+        after: Optional[str] = None,
+        accept_datetime: Optional[str] = None,
+        fields: Optional[List[Union[str, LabelFields]]] = None,
+        **kwargs,
+    ) -> ItemPaged[str]:
+        """Gets a list of labels.
+
+        :keyword name: A filter for the name of the returned labels. Default value is None.
+        :paramtype name: str or None
+        :keyword after: Instructs the server to return elements that appear after the element referred to
+            by the specified token. Default value is None.
+        :paramtype after: str or None
+        :keyword accept_datetime: Requests the server to respond with the state of the resource at the
+            specified time. Default value is None.
+        :paramtype accept_datetime: str or None
+        :keyword fields: Used to select what fields are present in the returned resource(s). Default
+            value is None.
+        :paramtype fields: list[str or ~azure.appconfiguration.models.LabelFields] or None
+        :return: An iterator of labels.
+        :rtype: ~azure.core.paging.ItemPaged[str]
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+        """
+        try:
+            return self._impl.get_labels(  # type: ignore[return-value]
+                name=name,
+                after=after,
+                accept_datetime=accept_datetime,
+                select=fields,
+                cls=lambda objs: [x.name for x in objs],
+                **kwargs,
+            )
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")  # pylint: disable=raise-missing-from
```

Comment:
This is a very low level error. And the message seems wrong - there is no connection string secret passed in? And why are we disabling the pylint warning - it seems like it is telling you to do the right thing?

Reply:
I agree - I this we should strip all these clauses out.
If we really feel that adding the additional message to a connection string parsing error is necessary, this try/except should be moved to wherever that error is actually being raised, not added to every method.